File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dic = {}
    for elem in participant:
        if elem in dic:
            dic[elem] += 1
        else:
            dic[elem] = 1
    for elem in completion:
        dic[elem] -= 1
    for key, value in dic.items():
        if value == 1:
            answer += key
            break
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Set;

class Solution {
    
    public String solution(String[] participant, String[] completion) {
      String answer;
      HashMap<String, Integer> h = new HashMap<>();

      for(String elem : participant) {
        h.put(elem, h.getOrDefault(elem, 0) + 1);
      }

      for(String elem : completion) {
        h.replace(elem, h.get(elem) - 1);
        if(h.get(elem).equals(0)) {
          h.remove(elem);
        }
      }

      answer = h.keySet().iterator().next();
      return answer;
    }
}